flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/files/StaticFileServerHandler.java

```diff
@@ -234,8 +237,17 @@ private void respondAsLeader(ChannelHandlerContext ctx, HttpRequest request, Str
 							if (!rootURI.relativize(requestedURI).equals(requestedURI)) {
 								logger.debug("Loading missing file from classloader: {}", requestPath);
 								// ensure that directory to file exists.
-								file.getParentFile().mkdirs();
-								Files.copy(resourceStream, file.toPath());
+								if (!file.getParentFile().mkdirs()) {
+									throw new IOException("Could not create directories for file " + file);
+								}
+								synchronized (COPY_LOCK) {
```

Comment:
Because this was easy to write, is easy to review, does not change any behavior and isn't particularly intrusive.

I agree that we should have a discussion as to whether we should load the files lazily or not, but not now when the next release is coming up and everyone is scrambling to fix the most issues in as little time as possible